### DISCUSSION TITLE
Canvas Collections times out after 120 seconds?

### DISCUSSION THREAD

**Discussion by omachado-id** [djplaner/canvas-collections#284] [Q&A]
Hi,
I've just downloaded Canvas Collections (CC) into my Chrome browser (ver. 116.0.5845.111), after successfully installing the stable version of Tampermonkey.  All that worked well, and in trying out CC, it seems like a very useful tool.

However, a nagging feature (or bug?) that I've encountered is that CC keeps timing out while editing is ON.  It repeatedly shuts editing to OFF automatically after a very very short time, displaying a dialog that states it's doing so because of 120 seconds of inactivity.  So far, I've found this aspect the most difficult to deal with regarding CC.

Q: Is there any way to modify or shut OFF this auto-shutdown.  I find it very difficult to explore the features while editing, let along find content to edit with CC, when it keeps shutting down editing?  If there is no way to edit or shut that down, could you consider adding a way to do this for future versions of Canvas Collections.  Any tips with this issue will be appreciated.

**Comment by djplaner** [comment]
G'day,

Welcome. You're the first discussant.  Good to hear you see some utility in Collections. Hope this helps.

The time out is actually intended as a feature (more below), but in your context it is certainly acting as a bug.  

As a quick solutions, I've just bumped this timeout to 10 minutes.  If you reload [the Collections userscript](https://github.com/djplaner/canvas-collections/raw/main/dist/canvas-collections.user.js) you should benefit from that increase (the userscript will eventually auto-update with this change, depending on your setting)

A workaround you could use, is to make a small change (e.g. add a letter to a module description) ever 2 (or now 10 minutes). If Collections detects a change it refreshes the timeout to 0.

## It is a feature?

At least in terms of it has been designed into Collections. But it's also a kludge. 

All the data Collections keeps about your course is saved into a Canvas page within your course. Not into a database, which would be the better solution if the Collections was part of Canvas or a LTI with it's own database.  A Canvas page works, but there's can be problems if more than one person tries to modify the Collections data at the same time. Data lost or overwritten

The "solution" Collections uses is an edit lock. Only one person can edit at a time. Which is why you have to turn `edit on`. To prevent someone hogging the edit lock, Collections will timeout the edit lock if you've not made any changes within a fixed time period.

That time period was 120 seconds. It's now 10 minutes.

David.

**Reply by omachado-id** [reply]
Hi David,
Well, first, kudos to you for creating what is an increasingly impressive resource.  As a course designer, this may open up capabilities that my institution's current instance of Canvas doesn't offer the typical user.  I'm going to continue to work with it to build up familiarity, then bring to the attention of others for adoption at enterprise level (that's not my call but so far I'm impressed).

Second, I appreciate your attention to my question.  Ten minutes is definitely a more workable span before a time-out.  I've been doing one step, then saving, then another, then saving and so on, in order to get in under the 2 minutes, so 10 minutes will overcome this niggling issue.  I appreciate it!  I'll update the userscript as you've described, as I was just working in CC when I received your response.

Keep up the great work; you're on to something!

O.M.
